.github/workflows/automatic-release-to-main-merger.yml

```diff
@@ -84,15 +85,24 @@ jobs:
                 These release-merge-PRs are currently open: https://github.com/${{ github.repository }}/pulls?q=is%3Aopen+label%3A${{ env.LABEL_TYPE }}+head%3Amerge-${{ steps.get-branch-name.outputs.release_branch }}-main"
 
       - name: Create new branch 🐣
+        id: create-new-branch
         if: always()
         uses: peterjgrainger/action-create-branch@64aa569aea81305305c6e92bd236d8c427debff8
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           branch: ${{ steps.get-branch-name.outputs.new_branch }}
 
+      - name: Notify Slack if creating a new branch failed 💬
+        if: ${{ failure() && steps.create-new-branch.conclusion == 'failure' }}
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_INFRASTRUCTURE_SQUAD_MONITORS_WEBHOOK_URL }}
+        uses: Ilshidur/action-slack@689ad44a9c9092315abd286d0e3a9a74d31ab78a
+        with:
+          args: "🚨 There was a problem with creating a new branch containing commits from the ${{ env.RELEASE_BRANCH }} release branch. Check out the Github action: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+
       - name: Open pull request ☄️
-        if: always()
+        if: ${{ always() && steps.create-new-branch.conclusion == 'success' }}
```

Comment:
Would it be more robust to use `outcome` instead of `conclusion` (see differences [here](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context) )? We're not using `continue-on-error`.

Reply:
but why do we have this `always()` condition at all? Should the workflow stop at the first failure? It seems like the workflow here is branching out at "Close outdated release-merge PRs 🧹". Could we do this step at the very end, for instance?

Reply:
I've had to add the `always()` because an if check that uses an expression will always check the status check `success()` of previous steps by default so this wouldn't pass. If the `Close outdated release-merge PRs` were to happen at the end, won't it also close the newly formed PR? I think not, but doing a sanity check.

Reply:
Introduced this [new command](https://github.com/RasaHQ/rasa/pull/12293/files#diff-920c4f57f26132498041e27b49c19a7cdbe3bb2928c414dec98856be8b62c763R98-R99) to filter the newly created PR from the list of PRs that get closed. Let me know what you think.